src/decorator/Session.ts

```diff
@@ -1,20 +1,39 @@
-import {getMetadataArgsStorage} from "../index";
+import { getMetadataArgsStorage } from "../index";
+import { ParamOptions } from "../decorator-options/ParamOptions";
+
 
 /**
  * Injects a Session object to the controller action parameter.
  * Must be applied on a controller action parameter.
  */
-export function Session(objectName?: string): Function {
+export function Session(options?: ParamOptions): ParameterDecorator;
+/**
+ * Injects a Session object to the controller action parameter.
+ * Must be applied on a controller action parameter.
+ */
+export function Session(propertyName: string, options?: ParamOptions): ParameterDecorator;
+
+export function Session(optionsOrObjectName?: ParamOptions|string, paramOptions?: ParamOptions): ParameterDecorator {
+    let propertyName: string|undefined;
+    let options: ParamOptions|undefined;
+    if (typeof optionsOrObjectName === "string") {
+        propertyName = optionsOrObjectName;
+        options = paramOptions || {};
+    } else {
+        options = optionsOrObjectName || {};
```

Comment:
@19majkel94 based on the conversation in #140 I thought we decided it didn't make sense to support `@Session({ required: false})`, which I think this would allow you to do now.